internal/proto/volumegroup.proto

```diff
@@ -0,0 +1,106 @@
+      returns (ControllerGetVolumeGroupResponse) {}
+}
+// CreateVolumeGroupRequest holds the required information to
+// create a volume group
+message CreateVolumeGroupRequest {
+  // suggested name for volume group (required for idempotency)
+  // This field is REQUIRED.
+  string name = 1;
+
+  // params passed to the plugin to create the volume group.
+  // This field is OPTIONAL.
+  map<string, string> parameters = 2;
+
+  // Secrets required by the driver to complete the
+  // create request.
+  string secret_name = 3;
+  string secret_namespace = 4;
+
+  // Specify volume_ids that will be added to the volume group.
+  // This field is OPTIONAL.
+  repeated string volume_ids = 5;
+}
+// CreateVolumeGroupResponse holds the information to send when
+// volumeGroup is successfully created.
+message CreateVolumeGroupResponse {
+  // Contains the identifier of the newly created volume group.
+  // This field is REQUIRED.
+  string volume_group_id = 1;
+}
+// DeleteVolumeGroupRequest holds the required information to
+// delete a volume group
+message DeleteVolumeGroupRequest {
+  // The ID of the volume group to be deleted.
+  // This field is REQUIRED.
+  string volume_group_id = 1;
+
+  // Secrets required by the driver to complete the
+  // delete request.
+  string secret_name = 2;
+  string secret_namespace = 3;
+}
+// DeleteVolumeGroupResponse holds the information to send when
+// volumeGroup is successfully deleted.
+message DeleteVolumeGroupResponse {
+}
+// ModifyVolumeGroupMembershipRequest holds the required
+// information to modify a volume group
+message ModifyVolumeGroupMembershipRequest {
+  // The ID of the volume group to be modified.
+  // This field is REQUIRED.
+  string volume_group_id = 1;
+
+  // Specify volume_ids that will be in the modified volume group.
+  // This list will be compared with the volume_ids in the existing
+  // group.
+  // New ones will be added and missing ones will be removed.
+  // If no volume_ids are provided, all existing volumes will
+  // be removed from the group.
+  // This field is OPTIONAL.
+  repeated string volume_ids = 2;
+
+  // Secrets required by the driver to complete the
+  // modification request.
+  string secret_name = 3;
+  string secret_namespace = 4;
+}
+// ModifyVolumeGroupMembershipResponse holds the information to
+// send when volumeGroup is successfully modified.
+message ModifyVolumeGroupMembershipResponse {
+}
+// ControllerGetVolumeGroupRequest holds the required
+// information to get information on volume group
+message ControllerGetVolumeGroupRequest {
+  // The ID of the volume group to fetch current volume group
+  // information for.
+  // This field is REQUIRED.
+  string volume_group_id = 1;
+
+  // Secrets required by the driver to complete the
+  // get request.
+  string secret_name = 2;
+  string secret_namespace = 3;
+}
+// ControllerGetVolumeGroupResponse holds the information to
+// send when volumeGroup information was successfully gathered.
```

Comment:
this comment does not make much sense anymore now

Reply:
Looking at other proto files, where we return an empty response. We are following a similar convention there as well, so I think we can let it be like this.

Reply:
the (empty) contents of the message is fine, the comment `ControllerGetVolumeGroupResponse holds the information ..` does not, it does not hold any information at all. Similar for the `*Response` messages above.